### ISSUE TITLE
KSQL Deployment issue after adding value_schema_id in the stream creation.

### ISSUE THREAD

**Issue by avinch45** [confluentinc/ksql#9082]
**Is your feature request related to a problem? Please describe.**
The challenge we have now for KSQL deployment after adding value_schema_id attribute is, middleware team could not execute the KSQL files from QA migration folders to Production, because we have Schema id values included in the KSQLs which would be different for the topics, because schema registries are different for these environments.

**Development concerns,** 
- We need to have Schema ids beforehand In the production environment to include them in the KSQL files and create Middleware tickets.
- Also we have to create and maintain two sets of files one for Dev/QA and other for Production.

Deployment team 
- They could not follow the process suggested by confluent , where they were supposed to execute the same files in QA and production with out any manual intervention, But they are unable to do it.

**Describe the solution you'd like**
We were expecting a process for deployment of KSQL's with out any dependencies with environments and clusters, where i can execute my code anywhere , irrespective of cluster/ environment we deploy. ( Like KStreams Java code )

**Describe alternatives you've considered**
We tried an alternative like below 
We can include a Get request (below) in a shell script to get the latest versions schema id for the Topic and update the schema id field dynamically in the specified KSQL file from the scripts.

Change in process would be, Deployment team has to execute the above shell script and take the KSQL file generated after dynamically populating the schema id. copy this file and execute the same way how we are doing it now.

schema_id=$(curl -s  -X GET -u $sr_auth $sr_url/subjects/$schema_name/versions/latest/ | jq ' .id')

Pros: The files provided from Kafka development would be same there is no Manual intervention.
Cons: Files will be different in QA and Prod in this approach also as the schema ids are different.

**Additional context**
We added this value_schema_id in the KSQL code for creating streams as the schema is getting evolved automatically while creating streams.

**Comment by colinhicks** [comment]
@avinch45, as you noted, the use of schema IDs does create a dependency on a separate system: Schema Registry, and there are several implications for referring to them in a KSQL workflow that uses multiple environmental tiers:
* For a given Schema Registry cluster, schema IDs are globally unique and monotonically increasing.
* For any two Schema Registry clusters, an identical schema stored to each cluster is not guaranteed to have the same schema ID.
* Using the `value_schema_id` or `key_schema_id` property tells KSQL to use a specific version, by ID, of the schema subject. For a given schema subject, KSQL cannot guess the desired version or query the latest ID when using either of these properties.

For these reasons, it's expected that schema IDs would need to be provided to KSQL as query configuration, and it's also expected that different KSQL environments (e.g. dev, test, and prod) may require different values for these configurations.

That said, it's still a best practice to use the same query files across different KSQL environments, even when using schema IDs. You can leverage [variable substitution](https://docs.ksqldb.io/en/latest/how-to-guides/substitute-variables/) to define common SQL, using placeholders for schema IDs.

```sql
CREATE STREAM mystream1  WITH (
  kafka_topic = 'stream1',
  value_format = 'avro',
  value_schema_id = ${mystream1_schema_id}
);
```

You can then provide the correct value per KSQL environment by supplying the respective variable value when you submit the SQL to each KSQL environment. For example in the REST API:
```sh
curl --http1.1 \
     -X POST "http://my-dev-server:8088/ksql" \
     -d $'{
  "ksql": "<contents of file above>",
  "streamsProperties": {},
  "sessionVariables": {"mystream1_schema_id": 10001}
}' 

curl --http1.1 \
     -X POST "http://my-prod-server:8088/ksql" \
     -d $'{
  "ksql": "<contents of file above>",
  "streamsProperties": {},
  "sessionVariables": {"mystream1_schema_id": 9999}
}' 
```

Variable substitution can also be applied by the Java client and the [migrations tool](https://docs.ksqldb.io/en/latest/operate-and-deploy/migrations-tool/) when submitting queries.

**Comment by avinch45** [comment]
@colinhicks  Yes , we did considered this option, But initially confluent team helped our middleware team to set up the build process in such a away that , they were supposed to execute the KSQL files we share with them, if this is the process they have to follow few extra steps like executing the scripts to for replacing the schema id with the latest id in those KSQL files we share them. 

Also we had a dependency here, where Schema of the topics is owned by a different team, So we have to wait for their deployment and get then do ours or we should go hand in hand.

Citing all these problems and dependencies, we raised this issue , if there is any other solution please let us know. 

@Adedoyin24 , Also know the context, he was out POC from confluent

Thank you !
